src/VivViewer.js

```diff
@@ -2,43 +2,145 @@ import React, { PureComponent } from 'react';
 import DeckGL from '@deck.gl/react';
 import { OrthographicView } from '@deck.gl/core';
 import { VivViewerLayer, StaticImageLayer } from './layers';
+import VivViewerOverview from './VivViewerOverview';
 
 export default class VivViewer extends PureComponent {
+  constructor(props) {
+    super(props);
+    const {
+      initialViewState,
+      overview,
+      viewWidth,
+      viewHeight,
+      loader
+    } = this.props;
+    const vivOverview = overview
+      ? new VivViewerOverview({
+          ...overview,
+          viewWidth,
+          viewHeight,
+          loader,
+          offset: overview.offset,
+          overviewScale: overview.scale
+        })
+      : null;
+    this.state = {
+      viewState: {
+        detail: { ...initialViewState, id: 'detail' },
+        overview: overview ? vivOverview.getViewState(initialViewState) : null
+      },
+      overview: vivOverview
+    };
+    this._onViewStateChange = this._onViewStateChange.bind(this);
+    this.layerFilter = this.layerFilter.bind(this);
+  }
+
+  // This prevents only the `draw` call of a layer from firing,
+  // but not other layer lifecycle methods.  Nonetheless, it is
+  // still useful.
+  // eslint-disable-next-line class-methods-use-this
+  layerFilter({ layer, viewport }) {
+    if (layer.id.includes(viewport.id)) {
+      // Draw the static layer in the overview
+      return true;
+    }
+    return false;
+  }
+
+  _onViewStateChange({ viewId, viewState }) {
+    // Save the view state and trigger rerender
+    // only for changes to the `detail` view
+    if (viewId === 'detail') {
+      const { overview } = this.state;
+      const newViewState = {};
+      newViewState.detail = viewState;
+      newViewState.overview = overview
+        ? overview.getViewState(viewState)
+        : null;
+      this.setState({ viewState: newViewState });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { overview, viewWidth, viewHeight, loader } = this.props;
+    if (prevProps.overview !== overview) {
+      const vivOverview = new VivViewerOverview({
+        ...overview,
+        viewWidth,
+        viewHeight,
+        loader,
+        offset: overview.offset,
+        overviewScale: overview.scale
+      });
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState(prevState => {
+        const newViewState = { ...prevState.viewState };
+        newViewState.overview = vivOverview.getViewState(
+          prevState.viewState.detail
+        );
+        return { viewState: newViewState, overview: vivOverview };
+      });
+    }
+  }
```

Comment:
Ah I think this is the issue. If `prevProps.overview` is a config, and in the new props is `overview === null`, this will throw because `overview.offset` and `overview.scale` do no exist on `null`.

Reply:
We currently handle a on-way toggle, `no overview ==> overview` but not `overview ==> no overview`